src/spells.cpp

```diff
@@ -637,6 +645,12 @@ bool Spell::playerSpellCheck(Player* player) const
 		return false;
 	}
 
+	if (player->getHealth() < static_cast<int32_t>(getHealthCost(player)) && !player->hasFlag(PlayerFlag_HasInfiniteMana)) {
+		player->sendCancelMessage(RETURNVALUE_NOTENOUGHMANA);
```

Comment:
needs a new return value, so it shows the correct message

Reply:
fixed, added 'not enought hp' msg